sql-plugin/src/main/320/scala/com/nvidia/spark/rapids/shims/v2/TypeSig320.scala

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.nvidia.spark.rapids.shims.v2
+
+import ai.rapids.cudf.DType
+import com.nvidia.spark.rapids.{TypeEnum, TypeSig}
+
+import org.apache.spark.sql.types.{DataType, DayTimeIntervalType, YearMonthIntervalType}
+
+/** TypeSig for Spark 3.2.0+ which adds DayTimeIntervalType and YearMonthIntervalType support */
+final class TypeSig320(
```

Comment:
I see two possibilities on how to do this

1. We do something like this were we have a TypeSigBase in the common code and then an actual TypeSig implementation class in shim layers. The problem with this is `TypeSig.all` and the other Spark helper values will not reflect "all" of the types. To do that we would need to move all or part of object TypeSig into the shim layer, or we would have to have the overrides for each of the things that touch `TypeSig.all` etc go into the shim layer.
2. We turn it around and have a static TypeSigUtil object in the shim layer that TypeSig calls into to do checks with new types.

For me I am  leaning towards the second approach because it is consistent between the static and non-static code.

Reply:
Thx @revans2, I just had a new commit for the second approach. Could you help to take a look at it? Thx